IntelFsp2Pkg/FspSecCore/SecFsp.c

```diff
@@ -68,11 +68,11 @@ SecGetPlatformData (
   FspPlatformData->CodeRegionBase      = 0;
   FspPlatformData->CodeRegionSize      = 0;
 
-  //
-  // Pointer to the size field
-  //
-  TopOfCar = PcdGet32 (PcdTemporaryRamBase) + PcdGet32 (PcdTemporaryRamSize);
-  StackPtr = (UINT32 *)(TopOfCar - sizeof (UINT32));
+  if ((TopOfCar == 0) || (TopOfCar == 0xFFFFFFFF)) {

```

Comment:
If above 4G is not supported for temp ram base, please consider changing it as below.
if ((TopOfCar == 0) || (TopOfCar >= MAX_UINT32)) {